src/com/mishiranu/dashchan/ui/navigator/page/PostsPage.java

```diff
@@ -371,8 +371,19 @@ protected void onCreate() {
 				replyable, postStateProvider, getFragmentManager(),
 				recyclerView, retainableExtra.postItems, retainableExtra.hiddenPosts);
 		recyclerView.setAdapter(adapter);
-		recyclerView.addItemDecoration(new DividerItemDecoration(recyclerView.getContext(),
-				(c, position) -> adapter.configureDivider(c, position).horizontal(dividerPadding, dividerPadding)));
+		DividerItemDecoration divider = new DividerItemDecoration(recyclerView.getContext(),
+				(c, position) -> adapter.configureDivider(c, position).horizontal(dividerPadding, dividerPadding));
+		divider.setSkipCallback(position -> {
+			if (Preferences.isHighlightUserPosts()) {
+				if (adapter.getConfigurationSet().postStateProvider.isUserPost(adapter.getItem(position).getPostNumber()))
+					return true;
+				if ((position + 1) < adapter.getItemCount()) {
+					return adapter.getConfigurationSet().postStateProvider.isUserPost(adapter.getItem(position + 1).getPostNumber());
```

Comment:
What does this part do?

Reply:
Checking whether the current post is marked as user post or whether the next one is so. Necessary to remove the divider if the highlight feature is on

Reply:
Is checking the next post necessary? As far as i understand the callback will be called for each post anyway so the callback should check the current post only.

Reply:
It seems that yes... when the post is highlighted, the gray dividing lines can be seen both at the top of the post and at the bottom. If we leave only the check for the current postItem, this will get rid of the line at the bottom, but leaves the line at the top as is.

Reply:
Now i understand. This if for the case when the current post is not a user post and the next post is the user's post. In this case the divider from the current post will be on top of the user's post and this will look bad.